modules/pam_lastlog/pam_lastlog.c

```diff
@@ -427,13 +436,44 @@ last_login_write(pam_handle_t *pamh, int announce, int last_fd,
         sleep(LASTLOG_IGNORE_LOCK_TIME);
     }
 
+    /*
+     * Failing to set the 'fsize' limit is not a fatal error. We try to write
+     * lastlog anyway, under the risk of dying due to a SIGXFSZ.
+     */
+    D(("setting limit for 'fsize'"));
+    getrlimit_res = getrlimit(RLIMIT_FSIZE, &old_limit);
+    if (getrlimit_res == 0) {
+	setrlimit_res = setrlimit(RLIMIT_FSIZE, &no_limit);
```

Comment:
Could you please also check whether the current limit is infinity and in that case avoid calling setrlimit()?